src/libs/meta/meta.c

```diff
@@ -1211,7 +1211,7 @@ int elektraSortTopology (KeySet * ks, Key ** array)
 					retVal = -1;
 					break;
 				}
-				else if (i == j)
+				else if ((long)i == (long)j)
 				{
```

Comment:
(Btw. if it is expensive for topology sort to have missing deps then it would of course better to remember missing deps and add them when needed)